buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchCluster.java

```diff
@@ -259,6 +264,10 @@ public void start() {
             // Can only configure master nodes if we have node names defined
             if (nodeNames != null) {
                 if (node.getVersion().onOrAfter("7.0.0")) {
+                    node.defaultConfig.keySet().stream()
```

Comment:
I'm pretty sure that, with the exception of `initial_master_nodes` all this can, and probably should, go in `ElasticsearchNode#start()`.

Reply:
Technically yes, but these are settings relating to how the cluster forms that relate to each-other,
so the settings are per node, but relate to the cluster itself. 
Granted we could pull other such settings up to the cluster level. 
I agree on your points on refactoring but would like to complete the back-port first.